runtime/interface.go

```diff
@@ -41,55 +43,130 @@ type Block struct {
+// methods with power of edits have caller as an argument which can be used
+// to limit access to these method calls from smart contracts only, transaction only, or
+// specific smart contracts (e.g. service account).
+type Accounts interface {
+	// NewAccount creates a new account address and set the exists flag for this account
+	NewAccount(caller Location) (address Address, err error)
+	// AccountExists returns true if the account exists
+	AccountExists(address Address) (exists bool, err error)
+	// NumberOfAccounts returns the number of accounts
+	NumberOfAccounts(caller Location) (count uint64, err error)
+
+	// SuspendAccount suspends an account (set suspend flag to true)
+	SuspendAccount(address Address, caller Location) error
+	// UnsuspendAccount unsuspend an account (set suspend flag to false)
+	UnsuspendAccount(address Address, caller Location) error
+	// returns true if account is suspended
+	IsAccountSuspended(address Address) (isSuspended bool, err error)
+}
+
+// AccountContracts manages contracts stored under accounts
+type AccountContracts interface {
+	// AccountContractCode returns the code associated with an account contract.
+	ContractCode(address AddressLocation) (code []byte, err error)
+	// UpdateAccountContractCode updates the code associated with an account contract.
+	UpdateContractCode(address AddressLocation, code []byte, caller Location) (err error)
+	// RemoveContractCode removes the code associated with an account contract.
+	RemoveContractCode(address AddressLocation, caller Location) (err error)
+	// Contracts returns a list of contract names under this account
+	Contracts(address AddressLocation, caller Location) (name []string, err error)
+}
 
-type Interface interface {
-	// ResolveLocation resolves an import location.
-	ResolveLocation(identifiers []Identifier, location Location) ([]ResolvedLocation, error)
+// AccountStorage stores and retrives account key/values
+type AccountStorage interface {
+	// Value gets a value for the given key in the storage, owned by the given account.
+	Value(key StorageKey, caller Location) (value StorageValue, err error)
+	// SetValue sets a value for the given key in the storage, owned by the given account.
+	SetValue(key StorageKey, value StorageValue, caller Location) (err error)
+	// ValueExists returns true if the given key exists in the storage, owned by the given account.
+	ValueExists(key StorageKey, caller Location) (exists bool, err error)
+	// StoredKeys returns list of keys and their sizes owned by this account
+	StoredKeys(address Address, caller Location) (iter StorageKeyIterator, err error)
+	// StorageUsed gets storage used in bytes by the address at the moment of the function call.
+	StorageUsed(address Address, caller Location) (value uint64, err error)
+	// Note: StorageCapacity has been moved to injected methods (similar to get balance)
+}
+
+// AccountKeys manages account keys
+type AccountKeys interface {
```

Comment:
Might be nice to add a count function? This would allow Cadence programs to iterate over all keys of an account.